pkg/apis/troubleshoot/v1beta2/collector_shared.go

```diff
@@ -27,6 +27,14 @@ type ClusterResources struct {
 	IgnoreRBAC    bool     `json:"ignoreRBAC,omitempty" yaml:"ignoreRBAC"`
 }
 
+type CPUMemUsage struct {
+	CollectorMeta `json:",inline" yaml:",inline"`
+	// How far back to query metrics in days.
+	Duration int `json:"duration,omitempty" yaml:"duration,omitempty"`
+	// How often to sample metrics in hours.
+	SamplingStep int `json:"samplingStep,omitempty" yaml:"samplingStep,omitempty"`
+}
+
```

Comment:
Have you checked some existing data/serialization formats for metrics? Maybe from prometheus package itself. It would be great for interoperability to use some known format.